src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+package org.jitsi.impl.neomedia.rtp.remotebitrateestimator;
+
+import org.ice4j.util.RateStatistics;
+import org.jitsi.impl.neomedia.codec.audio.silk.K2aFLP;
+import org.jitsi.impl.neomedia.transform.PacketTransformer;
+import org.jitsi.service.neomedia.RawPacket;
+import org.jitsi.service.neomedia.rtp.RemoteBitrateEstimator;
+import org.jitsi.util.Logger;
+
+import javax.media.rtp.RTPHeader;
+import java.util.*;
+
+/**
+ * webrtc.org abs_send_time implementation as of June 26, 2017.
+ * @author Julian Chukwu
+ */
+public class RemoteBitrateEstimatorAbsSendTime
+    implements RemoteBitrateEstimator
+{
+    private static final Logger logger = Logger
+            .getLogger(RemoteBitrateEstimatorAbsSendTime.class);
+
+    private enum probeResult
+    {
+        kBitrateUpdated(0),
+        kNoUpdate(1);
+        int value;
+        probeResult(int x)
+        {
+            this.value = x;
+        };
+    }
+
+    private final Object critSect = new Object();
+
+    public final static int kTimestampGroupLengthMs = 5;
+    public final static int kAbsSendTimeFraction = 18;
+    public final static int kAbsSendTimeInterArrivalUpshift = 8;
+    public final static int kInterArrivalShift
+            =  kAbsSendTimeFraction + kAbsSendTimeInterArrivalUpshift;
+    public final static int kInitialProbingIntervalMs = 2000;
+    public final static int kMinClusterSize = 4;
+    public final static int kMaxProbePackets = 15;
+    public final static int kExpectedNumberOfProbes = 3;
+
+    private static final double kTimestampToMs = 1000.0 /
+            (1 << kInterArrivalShift) ;
+
+    public <K,V> List<K> Keys(TreeMap<K,V> _map)
+    {
+        ArrayList<K> keys = new ArrayList<K>();
+        for(Map.Entry<K,V> entry : _map.entrySet())
+        {
+            keys.add(entry.getKey());
+        }
+        return keys;
+    }
+
+    long ConvertMsTo24Bits(long timeMs)
+    {
+        long time24Bits = (long)(((timeMs << kAbsSendTimeFraction) + 500) /
+                1000) & 0x00FFFFFF;
+        return time24Bits;
+    }
+
+    /**
+      * Confirm this on line 133 of remote_bitrate_estimator_abs_send_time.h
+     */
+    public TreeMap<Long,Long> ssrcs_ = new TreeMap<Long, Long>();
+    public static ArrayList<Probe> probes_ = new ArrayList<>();
+    public static long totalProbesReceived;
+    public static  long firstPacketTimeMs;
+    public static long lastUpdateMs;
+    public static boolean umaRecorded;
+
+    /**
+     * @Todo confirm these
+     * clock and observer_ are defined as const
+     * e.g const Clock* const clock_;
+     * find out if there is need to make it final
+     */
+    public static long clock;
+    public static RemoteBitrateObserver observer_;
```

Comment:
In java "static" defines a class field instead of an instance field. These should not be static. Try to make them final, if something prevents it leave it out.